mne/forward/forward.py

```diff
@@ -1234,10 +1234,13 @@ def compute_depth_prior(forward, info, is_fixed_ori=None,
 
 def _stc_src_sel(src, stc):
     """Select the vertex indices of a source space using a source estimate."""
-    if isinstance(stc, VolSourceEstimate):
+    if isinstance(stc, list):
+        vertices = stc
+    elif isinstance(stc, VolSourceEstimate):
         vertices = [stc.vertices]
     else:
         vertices = stc.vertices
+    del stc
```

Comment:
_stc_src_sel() removes its input stc, but in _apply_forward() stc object is used after calling _stc_src_sel():

```
src_sel = _stc_src_sel(fwd['src'], stc)
if isinstance(stc, VolSourceEstimate):
    n_src = len(stc.vertices)
```

Reply:
The simulation code does not use `_apply_forward` -- I don't see a problem here. `_apply_forward` ensures that the number of vertices in `fwd` and `stc` match, whereas the simulation code is more tolerant, allowing the forward calculation to remove some of the vertices (e.g., due to the `mindist` when computing the forward).

Reply:
... oh wait, I think I see what you're saying. I'll add a test case for the stc having vertices not in the source space

Reply:
But is `_apply_forward` somewhere else? Will it raise an error?

Reply:
I think perhaps you're misunderstanding what `del stc` does here. All it does is remove the variable from the current scope (in this case, to make it clear it's not needed and ensure we don't errantly use it) -- it does not remove it from the scope of the caller (`_apply_forward`)

Reply:
Yes, I was misunderstanding what it did.